### ISSUE TITLE
verifyPhoneNumber Promise never fulfills on Android on v1.0.1.

### ISSUE THREAD

**Issue by cairinmichie** [chemerisuk/cordova-plugin-firebase-authentication#36]
I have a project in production that uses this method and I updated my dependancies and this broke. Any idea why it broke?

**Comment by cairinmichie** [comment]
And onAuthStateChanged is never called.

**Comment by DiegoParadeda** [comment]
@cairinmichie I had a similar situation to yours.
The first thing I did when looking up this plugin was looking for "onAuthStateChanged" inside it but I also couldn't find it. Still I added this plugin and "whitelist" plugin to my project and it worked.
I used phonegap to compile it but also did it from framework 7 using GUI bash. To do that either add the plugin one line declaration to you config.xml file (in this case I added to the one in root folder AND the config.xml in www folder) and the compiler will add the rest of the files for you.
Another way of adding them is through "cordova plugin add cordova-plugin-firebase --save" in your command line aiming for the root folder of your project.
Good luck!

**Comment by cairinmichie** [comment]
So my problem is that I want to login using js, and not in the platform code. I was using [this](https://github.com/chemerisuk/cordova-plugin-firebase-authentication/issues/20) workaround. But that doesn't work on v1.0.1 anymore because verifyPhoneNumber isn't really supported as far as I can tell. 

So I've resorted to using v0.13.2 because verifyPhoneNumber still works there. However this means I can't use the latest versions of @chemerisuk other plugins because of dependancy problems with cordova-support-google-services. So I'm kinda stuck at the moment.

**Comment by chemerisuk** [comment]
@cairinmichie Have you registered the `onAuthStateChanged` callback?

**Comment by cairinmichie** [comment]
Yes, that still doesn't support my use case though? Unless I'm missing something? Correct me if I'm wrong, but there's no way to send an sms if the Android auto-verify has been used?

**Comment by chemerisuk** [comment]
> there's no way to send an sms if the Android auto-verify has been used?
Yes, no way at present. But why do you need this?

**Comment by cairinmichie** [comment]
My problem is this:

I use [vuexfire](https://github.com/posva/vuexfire) extensively throughout my vuex store. Vuexfire requires that you use the firebase js library for authentication, as far as I'm aware. So I was using the `signInWithVerificationId` function to login my users with the js library rather than android/ios. I'm not sure there is a way to accomplish logging in with the js library when using `onAuthStateChanged`.
